client/python/thegame/experimental/gymbase.py

```diff
@@ -65,12 +65,12 @@ def reset(self):
         self.server.reset()
         self.server.tick()
         self.game_state = self.client.fetch_state()
-        return self.game_state_to_observation(self.game_state)
+        return self.game_state_to_observation(self.game_state, reset=True)
 
     def action_to_controls(self, action) -> Controls:
         raise NotImplementedError
 
-    def game_state_to_observation(self, game_state: GameState):
+    def game_state_to_observation(self, game_state: GameState, reset=False):
```

Comment:
How about putting the tick number or is_reset into the GameState object?
There might be more places to modify though.